packages/adminrouter/extra/src/README.md

```diff
@@ -264,28 +264,174 @@ it, but it does not seem to be justified. For example:
-
-In order for the forwarding to work reliably across task failures, we recommend co-locating the endpoints with the task. This way, if the task is restarted on a potentially other host and with different ports, Admin Router will pick up the new labels and update the routing. NOTE: Due to caching there might be an up to 30-second delay until the new routing is working.
-
-We would recommend having only a single task setting these labels for a given `service-name`.
-In the case of multiple task instances with the same `service-name` label, Admin Router will pick one of the tasks instances deterministically, but this might make debugging issues more difficult.
+Endpoint supports WebSockets, by default disables all NGINX buffering and all
+redirects from services are rewritten in order to support `/service` prefix
+clients are using.
+
+### Intended use
+It is very important to remember that `/service` endpoint was designed to work
+with root Marathon only. Tasks launched by MarathonOnMarathon instances will not
+be reachable via it.
+
+In the case of Mesos frameworks, this becomes a bit more complex - all the frameworks
+are visible in Mesos' `/state-summary` endpoint output so in theory it is possible
+to access a framework launched using MoM or by hand. Unfortunately, Mesos as
+of now does not enforce framework name uniqueness and thus it is possible to
+launch two frameworks with the same name. AR in such case will route in
+deterministic manner. Root Marathon by default enforces unique names for all the
+tasks and frameworks it has under control.
+
+### Operation
+Under the hood, `/service` endpoint follows non-trivial algorithm in order to
+resolve request path into IP + Port pair.
+
+#### Recursive resolving
+There is no way to tell which part of the path is namespace+service
+name and which one is just a resource address. For example, asumming that
+request path is `/foo/bar/baz/picture.jpg` we can have:
+* namespace+name: `foo`, resource address: `/bar/baz/picture.jpg`
+* namespace+name: `foo/bar`, resource address: `/baz/picture.jpg`
+* namespace+name: `foo/bar/baz`, resource address: `/picture.jpg`
+* namespace+name: `foo/bar/baz/picture.jpg`, resource address: `/`
+
+So the idea is to it iteratively try to resolve each "candidate" namespace+name
+and assume that first-match wins.
+
+Assuming that we deployed `foo/bar` Nginx task, we will see two iterations:
+* first one with `foo` candidate namespace+service, which is not present in the
+cluster
+* second one with `foo/bar` candidate namespace+servie name, which results in
```

Comment:
s/servie/service

Above there is "service", here there is "service name". Please make these consistent.